### PR TITLE
Jason/subscription components

### PR DESCRIPTION
**Pull request by duidae**
newsletter subscription page related components, [ticket](https://app.asana.com/0/1203699264568808/1203699266233793)

### CODE REVIEW

File: packages/react-components/src/button/stories/toggleButton.stories.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  component: ToggleButton,
+  argTypes: {
+    value: {
+      defaultValue: false,
+      options: [true, false],
+    },
+  },
+}
+
+const Template = args => <ToggleButton {...args} />
```
Comment (Aylie-Chou) [inline packages/react-components/src/button/stories/toggleButton.stories.js:15]: Question: toggle button on storybook cannot toggle by click component UI, is it as expected ?

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]
Mostly looking good.
I found one spec do not been implement, would the change of button components (icons on both left & right hands) be included in this patch ?
tmp

File: packages/react-components/src/text/headline.js [hunk 2]
```diff
@@ -67,89 +67,37 @@ const H6Container = styled(DefaultContainer)`
...
-    <H3Container fontFamily={fontFamily} className={className}>
-      {text}
-    </H3Container>
-  )
-}
-H3.propTypes = {
-  text: PropTypes.string,
-  type: TYPE_PROP_TYPES,
-  className: PropTypes.string,
+  return headerline
```
[the remark this answers is not in the document]
Comment (duidae) [reply packages/react-components/src/text/headline.js]: renamed to `headline`

File: packages/react-components/src/text/paragraph.js [hunk 3]
```diff
@@ -26,56 +28,33 @@ const P4Container = styled(defaultContainer)`
...
-  className: PropTypes.string,
-}
+const withContainer = ParagraphContainer => {
+  const paragraph = ({
+    text = '',
+    weight = WEIGHT.normal,
+    className = '',
+    ...props
+  }) => {
+    return (
```
[the remark this answers is not in the document]
Comment (duidae) [reply packages/react-components/src/text/paragraph.js]: removed.

File: packages/react-components/src/button/components/toggleButton.js [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+      transform: translate(20px, -50%);
+    }
+  }
+`
+
+const ToggleButton = ({
+  value = false,
+  label = ['', ''],
+  onChange = () => {},
+  ...props
```
[the remark this answers is not in the document]
Comment (duidae) [reply packages/react-components/src/button/components/toggleButton.js:68]: added `disabled` prop

File: packages/react-components/src/button/components/toggleButton.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    background: ${colorBrand.heavy};
+    &:before {
+      transform: translate(20px, -50%);
+    }
+  }
+`
+
+const ToggleButton = ({
+  value = false,
+  label = ['', ''],
```
[the remark this answers is not in the document]
Comment (duidae) [reply packages/react-components/src/button/components/toggleButton.js]: renamed to `labelOn` `labelOff`

File: packages/react-components/src/badge/index.js [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { P2 } from '../text/paragraph'
+
+const Container = styled.div`
+  background: white;
```
[the remark this answers is not in the document]
Comment (duidae) [reply packages/react-components/src/badge/index.js]: added color settings

File: packages/react-components/src/button/stories/toggleButton.stories.js (hunk 1 above)
Comment (duidae) [reply packages/react-components/src/button/stories/toggleButton.stories.js:15]: for simple stateless component, using prop for controlling is sufficient, we dont need to maintain extra inner state

**Comment by duidae** [comment]
> Mostly looking good. I found one spec do not been implement, would the change of button components (icons on both left & right hands) be included in this patch ? tmp

added change for textbutton, please have a look, thanks.

File: packages/react-components/src/badge/badge.stories.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
+import React from 'react'
+import Badge from '.'
+import { colorBrand } from '@twreporter/core/lib/constants/color'
+
+export default {
+  title: 'Badge/Badge',
```
[the remark this answers is not in the document]
Comment (duidae) [reply packages/react-components/src/badge/badge.stories.js]: removed.

File: packages/react-components/src/button/components/textButton.js [hunk 8]
```diff
@@ -69,22 +72,39 @@ const TextButton = ({
...
     >
-      {textJSX}
-      {iconComponent}
+      {components}
     </ButtonContainer>
   )
 }
 TextButton.propTypes = {
   iconComponent: PropTypes.element,
+  iconPosition: PropTypes.oneOf([Position.LEFT, Position.RIGHT]),
```
Comment (Aylie-Chou) [inline packages/react-components/src/button/components/textButton.js:101]: The icon could appear on both left & right side, and each side could have its own icon.

File: packages/react-components/src/button/components/textButton.js [hunk 9]
```diff
@@ -69,22 +72,39 @@ const TextButton = ({
...
   text: PropTypes.string,
   size: SIZE_PROP_TYPES,
+  colorSet: PropTypes.shape({
```
Comment (Aylie-Chou) [inline packages/react-components/src/button/components/textButton.js:104]: I suppose we control all the color set by `theme`.
Do we add extra color spec this time ?

**Review by Aylie-Chou — CHANGES_REQUESTED** [review]

File: packages/react-components/src/button/components/textButton.js (hunk 8 above)
Comment (duidae) [reply packages/react-components/src/button/components/textButton.js]: fixed.

File: packages/react-components/src/button/components/textButton.js (hunk 9 above)
Comment (duidae) [reply packages/react-components/src/button/components/textButton.js]: fixed with type prop.

**Review by Aylie-Chou — APPROVED** [review]
LGTM, just one more thing also mention in this [comment](https://github.com/twreporter/membership-fe/pull/658#discussion_r1127689158).
I think it's okay if we fix it in next PR.

**Pull request merged**